python/pyspark/sql/column.pyi

```diff
@@ -116,3 +123,11 @@ class Column:
     def __nonzero__(self) -> None: ...
     def __bool__(self) -> None: ...
     def contains(self, item: Any) -> Column: ...
+
+def _to_java_column(col: ColumnOrName) -> JavaObject: ...
```

Comment:
All missing type hints were [internal use](https://pep8.org/#descriptive-naming-styles) methods, but it still be used by other modules: https://github.com/apache/spark/blob/364d1eaf10f51c357f507325557fb076140ced2c/python/pyspark/pandas/spark/functions.py#L22

I think we should move these methods into pyspark.sql.utils or somewhere in future.